system/datatypes/settings.php

```diff
@@ -218,7 +218,28 @@ function register($key) {
 			notify('error', __('error-registration'));
 			return 'reload';
 		}
-
+	
+		/*
+		 Function: typesOptionList
+		 Returns an html option list with data types
+		 
+		 Parameters:
+		 $select - which option should be preselected
+		 $omit - which data types should be omitted
+		 */
+		function typesOptionList($select, $omit) {
+			$html = '';
+			foreach (scandir('system/datatypes') as $file) {
```

Comment:
I think that this does not match the way the new page dialog generates the list of possible page types, see index.php and the `$hyphaPageTypes` variable. It might be good to move the list of known page types to some more centralized position, rather than generating it inside `index.php` so it can be reused here. Doing so might also make fixing #146 easier perhaps.

Reply:
"some more centralized position," any suggestions?

Reply:
There is a global `$hyphaPageTypes` that is documented in `base.php`, so I guess that would be a good place. I can imagine a `hypha_getPageTypes()` function in that same file that returns that variable? Though it seems you might not just need the (class)names, but also the "titles" (display names) of the page types (which you now assume are equal, [but apparently are not](https://github.com/PlanBCode/hypha/blob/master/index.php#L146)).

Perhaps `$hyphaPageTypes` could be extended to be an array of `classname => displayname` (essentially what [`$dataTypeMtx` in `index.php`] already is), or the new `hypha_getPageTypes()` could be extended to load the display names on-demand?

Adding such a function is probably best done in a separate commit, since it is essentially an independent change.

Maybe it would also be good to move the code that fills this variable from `index.php` into `base.php` as well (e.g. into a `hypha_findPageTypes()` or something?). That might be out of scope, but if we include the display name stuff now, that might involve refactoring index.php anyway, so perhaps better to do this properly right away?

Reply:
done